lib/live_debugger_refactor/services/process_monitor/actions.ex

```diff
@@ -0,0 +1,54 @@
+defmodule LiveDebuggerRefactor.Services.ProcessMonitor.Actions do
+  @moduledoc """
+  This module provides actions for the ProcessMonitor service
+  """
+
+  alias LiveDebuggerRefactor.Bus
+  alias LiveDebuggerRefactor.API.LiveViewDebug
+  alias LiveDebuggerRefactor.CommonTypes
+  alias LiveDebuggerRefactor.Services.ProcessMonitor.GenServers.ProcessMonitor
+
+  alias LiveDebuggerRefactor.Services.ProcessMonitor.Events.LiveViewBorn
+  alias LiveDebuggerRefactor.Services.ProcessMonitor.Events.LiveViewDied
+  alias LiveDebuggerRefactor.Services.ProcessMonitor.Events.ComponentCreated
+  alias LiveDebuggerRefactor.Services.ProcessMonitor.Events.ComponentDeleted
+
+  @spec register_component_created(ProcessMonitor.state(), pid(), CommonTypes.cid()) ::
+          ProcessMonitor.state()
+  def register_component_created(state, pid, cid) do
```

Comment:
I think it can raise an exception so you should use `!`
```suggestion
  def register_component_created!(state, pid, cid) do
```

Similarly in below